index.js

```diff
@@ -187,7 +189,7 @@ Wire.prototype.handshake = function (infoHash, peerId, extensions) {
 Wire.prototype.choke = function () {
   if (this.amChoking) return
   this.amChoking = true
-  while (this.peerRequests.length) this.peerRequests.pop()
+  this.peerRequests.slice(0, this.peerRequests.length)
```

Comment:
`slice` doesn't change the array. It just returns a shallow copy while leaving the original array in tact.

I think you wanted `splice`. Or, how about we just do: `this.peerRequests = []`